### PR TITLE
Sed 2435 the artifact repository connector duplicates keywords

### PR DESCRIPTION
**Pull request by iegorov777**

### CODE REVIEW

File: step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/AbstractArtifactRepository.java [hunk 1]
```diff
@@ -180,9 +175,9 @@ protected void wrapPlanInTestCase(Plan plan, String testCaseName){
 
 	private static class FileAndPlans {
 		private final File artifact;
-		private final List<Plan> plans;
+		private final StepJarParser.PlansParsingResult plans;
```
Comment (jeromecomte) [inline step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/AbstractArtifactRepository.java:178]: Would rename this field

File: step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/AbstractArtifactRepository.java [hunk 2]
```diff
@@ -94,7 +94,7 @@ protected FileAndPlans getAndParseArtifact(Map<String, String> repositoryParamet
 		String[] excludedClasses = repositoryParameters.getOrDefault(PARAM_EXCLUDE_CLASSES, ",").split(",");
 		String[] excludedAnnotations = repositoryParameters.getOrDefault(PARAM_EXCLUDE_ANNOTATIONS, ",").split(",");
 
-		List<Plan> plans = parsePlans(artifact,libraries,includedClasses,includedAnnotations,excludedClasses,excludedAnnotations);
+		StepJarParser.PlansParsingResult plans = parsePlans(artifact,libraries,includedClasses,includedAnnotations,excludedClasses,excludedAnnotations);
```
Comment (jeromecomte) [inline step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/AbstractArtifactRepository.java:97]: would rename this variable

File: step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/AbstractArtifactRepository.java [hunk 3]
```diff
@@ -127,20 +127,15 @@ private Plan buildTestSetPlan(ExecutionContext context, Map<String, String> repo
...
 			plan.setVisible(false);
 
-			Collection<Function> testCaseFunctions = plan.getFunctions();
-
-			plan.setFunctions(testCaseFunctions);
-			functions.addAll(testCaseFunctions);
-
-			enrichPlan(context, plan);
+			plan.getFunctions().addAll(fileAndPlan.plans.getFunctions());
+     		enrichPlan(context, plan);
```
Comment (jeromecomte) [inline step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/AbstractArtifactRepository.java:138]: code formatting

File: step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/AbstractArtifactRepository.java [hunk 4]
```diff
@@ -180,9 +175,9 @@ protected void wrapPlanInTestCase(Plan plan, String testCaseName){
 
 	private static class FileAndPlans {
```
Comment (jeromecomte) [inline step-repositories/step-repositories-artifact/src/main/java/step/repositories/artifact/AbstractArtifactRepository.java:176]: This class could be renamed to better match with its new purpose

**Review by jeromecomte — DISMISSED** [review]
Approving with some minor remarks

**Review by david-stephan — APPROVED** [review]

**Pull request merged**